tests/notebooks/e2e-wine/e2e-wine-kfp-mlflow-seldon.ipynb

```diff
@@ -0,0 +1,1233 @@
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "assert os.path.exists(OUTPUT_PARQUET_FILE)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "9d06a68e",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:52:38.511665Z",
+     "iopub.status.busy": "2023-08-09T14:52:38.487331Z",
+     "iopub.status.idle": "2023-08-09T14:52:38.555444Z",
+     "shell.execute_reply": "2023-08-09T14:52:38.554350Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# workflow component\n",
+    "preprocess_op = kfp.components.create_component_from_func(\n",
+    "    func=preprocess,\n",
+    "    output_component_file=\"preprocess-component.yaml\",\n",
+    "    base_image=\"python:3.8.10\",\n",
+    "    packages_to_install=[\"pandas\", \"pyarrow\"],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9e2b5fed",
+   "metadata": {},
+   "source": [
+    "## Train Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "6c43576b",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:52:38.561183Z",
+     "iopub.status.busy": "2023-08-09T14:52:38.559947Z",
+     "iopub.status.idle": "2023-08-09T14:52:38.613285Z",
+     "shell.execute_reply": "2023-08-09T14:52:38.612010Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def training(file_path: kfp.components.InputPath(\"parquet\")) -> str:\n",
+    "    import os\n",
+    "    import mlflow\n",
+    "    import pandas as pd\n",
+    "\n",
+    "    from sklearn.linear_model import ElasticNet\n",
+    "    from sklearn.metrics import classification_report\n",
+    "    from sklearn.model_selection import train_test_split\n",
+    "    \n",
+    "    df = pd.read_parquet(file_path)\n",
+    "    \n",
+    "    target_column=\"quality\"\n",
+    "    train_x, test_x, train_y, test_y = train_test_split(\n",
+    "        df.drop(columns=[target_column]),\n",
+    "        df[target_column], test_size=.25,\n",
+    "        random_state=42, stratify=df[target_column]\n",
+    "    )\n",
+    "\n",
+    "    mlflow.sklearn.autolog()\n",
+    "    with mlflow.start_run(run_name=\"elastic_net_models\") as run:\n",
+    "        mlflow.set_tag(\"author\", \"kf-testing\")\n",
+    "        lr = ElasticNet(alpha=0.5, l1_ratio=0.5, random_state=42)\n",
+    "        lr.fit(train_x, train_y)\n",
+    "        model_dir = \"model\"\n",
+    "        mlflow.sklearn.log_model(lr, model_dir, registered_model_name=\"wine-elasticnet\")\n",
+    "        return f\"{run.info.artifact_uri}/{model_dir}\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "709a10b5",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-08-09T14:52:38.617296Z",
+     "iopub.status.busy": "2023-08-09T14:52:38.616499Z",
+     "iopub.status.idle": "2023-08-09T14:52:51.556017Z",
+     "shell.execute_reply": "2023-08-09T14:52:51.553302Z"
+    }
+   },
+   "outputs": [
```

Comment:
Do we need to include the outputs? Maybe just upload the notebook before running it?

Reply:
Since notebooks have presentational value as well, they are usually shared with outputs included as an example. That's why I generally try to have sensible, well-formatted outputs, that don't clutter the end notebook too much.

Is there something specific that's bothering you there?

Reply:
I see that most notebooks provided in [kubeflow/examples](https://github.com/kubeflow/examples/) don't include their outputs. They do say "output might look something like this" in cases where it might be helpful for user understanding.

Reply:
I'll update the notebook accordingly!

Reply:
> I see that most notebooks provided in [kubeflow/examples](https://github.com/kubeflow/examples/) don't include their outputs. They do say "output might look something like this" in cases where it might be helpful for user understanding.

Yeah, I would prefer something like this one.